### PR TITLE
Bump minimum PHP version to 7.4 for WooCommerce beta matrix only.

### PR DESCRIPTION
**Pull request by c-shultz**
Fixes failing GitHub actions for WooCommerce beta + PHP 7.3

#### Changes proposed in this Pull Request

Title: Remove PHP 7.3 from WooCommerce beta test matrix

Updates WooCommerce beta test matrix so minimum PHP version is 7.4 instead of 7.3. This will temporarily fix test failures, but a long term fix is needed right away because WooCommerce 8.2 releases October 10th which will re-break the tests against the latest WooCommerce.

#### Testing instructions

* Confirm that tests pass in GitHub actions
* Confirm that WooCommerce beta tests are still running

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7391` or branch name `fix/bump-min-php-wc-beta` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/bump-min-php-wc-beta) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: da263a64d6d4a78b9b2f0cf72f52ac29b8041591
- Build time: 2023-10-06 06:36:43 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/fix-bump-min-php-wc-beta [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Update automated test matrix to remove PHP 7.3 tests against WooCommerce beta.
```
Comment (RadoslavGeorgiev) [inline changelog/fix-bump-min-php-wc-beta]: For dev patches, you can skip the message and just add a comment, similar to this: https://github.com/Automattic/woocommerce-payments/blob/1b6d3519a8d1f6c0906b32e849bbfcd49d32652d/changelog/rpp-basic-process. By doing so, the entry will not be included in the release changelog, which makes sense in this case, as the change will not be released to merchants.

**Review by RadoslavGeorgiev — APPROVED** [review]
LGTM :shipit: 

Note: I’m checking this on mobile, but the change in the matrix makes sense, and all PHP tests are passing. This is a temporary fix regardless, so I think we can confidently ship it.

**Comment by c-shultz** [comment]
I opened https://github.com/Automattic/woocommerce-payments/issues/7392 to track the permanent fix for this.

**Review by deepakpathania — APPROVED** [review]
Changes LGTM, discussion in Slack about the permanent fix for additional context: p1696573909001409/1696001826.186669-slack-CGGCLBN58

**Pull request merged**